test/optical_flow.jl

```diff
@@ -0,0 +1,115 @@
+using Images, TestImages
+
+# Testing constants
+test_image = "mandrill"
+number_test_pts = 200
+precision = 0.1
+max_error_points_percentage = 10
+max_allowed_error = 1.5
+max_lost_points_percentage = 5
+
+function test_lk(number_test_pts::Int64, flow::Array{Coordinate{Float64}, 1}, x_flow::Float64, y_flow::Float64, status::BitArray{1}, err::Array{Float64, 1}, precision::Float64)
+    #Testing parameters
+    error_pts = 0
+    max_err = 0
+    total_err = 0
+    lost_points = 0
+
+    for i = 1:number_test_pts
+        if status[i]
+            if abs(flow[i].x - x_flow) > precision || abs(flow[i].y - y_flow) > precision
+                error_pts += 1
+            end
+            if err[i] > max_err
+                max_err = err[i]
+            end
+            total_err += err[i]
+        else
+            lost_points += 1
+        end
+    end
+    return error_pts, max_err, total_err, lost_points
+end
+
+@testset "lucas-kanade" begin
+
+    #Basic translations (Horizontal)
+    img1 = testimage(test_image)
+    img2 = similar(img1)
+    for i = 1:size(img1)[1]
+        for j = 4:size(img1)[2]
+            img2[i,j] = img1[i,j-3]
+        end
+    end
+    test_img1 = Gray.(img1)
+    test_img2 = Gray.(img2)
+    corners = imcorner(test_img1, method=shi_tomasi)
```

Comment:
Instead of introducing the `Coordinate` type, perhaps just use `SVector` directly.  So instead of storing an array of `Coordinate` types, you will store an array of `SVector` instead. An `SVector` is allocated on the stack, and for the types of operations that you will perform I suspect that it will lead to much faster code.

The code for creating  the array of coordinates can be written much more concisely as follows:
```julia
r, c, _ = findnz(corners)
corner_coords = map((ri, ci) -> SVector{2}(ri, ci), r, c)
pts = rand(corner_coords, (number_test_pts,))
```

I discovered this neat approach on[ line 58 of core.jl from ImageFeatures](https://github.com/JuliaImages/ImageFeatures.jl/blob/master/src/core.jl).

Reply:
Done